### PR TITLE
Implemented search endpoint without using query-to-mongo.

### PR DESCRIPTION
**Pull request by emmaodia**
Work done for https://github.com/blockstack/bounties/issues/3

### CODE REVIEW

**Comment by friedger** [comment]
> The PR must also in include a written explanation of how you fixed the issue.

Could you please explain a bit what you did?
This look works looks not complete. This only chances the server. The client library still throws the same error.

File: src/controllers/ModelsController.ts [hunk 1]
```diff
@@ -77,6 +77,29 @@ const makeModelsController = (
     });
   });
 
+  ModelsController.getAsync('/search/:search', async (req, res) => {
+    //A complex search query that returns the entered params of the search without using query-to-mongo
+    const { search } = req.params;
+    const options = { limit: req.query.limit, offset: req.query.offset };
+    // const mongo = queryToMongo(req.query.path, {
```
Comment (friedger) [inline src/controllers/ModelsController.ts]: Please remove commented code.

**Comment by emmaodia** [comment]
@friedger Code refactored. Kindly review. Thank you for your time.

**Comment by friedger** [comment]
Could you please explain how your solution solves the issue?

Emmanuel Odianosen .O. <notifications@github.com> schrieb am Mo., 20. Jan.
2020, 01:40:

File: src/controllers/ModelsController.ts [hunk 2]
```diff
@@ -77,6 +77,28 @@ const makeModelsController = (
     });
   });
 
+  ModelsController.getAsync('/search', async (req, res) => {
+    //A complex search query that returns the entered params of the search without using query-to-mongo
+    const { search } = req.body;
+    const mongo = queryToMongo(search, {
```
Comment (friedger) [inline src/controllers/ModelsController.ts]: queryToMongo does not support $or

**Comment by friedger** [comment]
Should be closed and work continued on #36

**Pull request closed without merging**